contracts/compliance/modular/modules/MaxBalanceModule.sol

```diff
@@ -105,6 +110,21 @@ contract MaxBalanceModule is AbstractModule {
         emit MaxBalanceSet(msg.sender, _max);
     }
 
+    /**
+     *  @dev pre-set the balance of a token holder per ONCHAINID
+     *  @param _compliance the address of the compliance contract to preset
+     *  @param _id the ONCHAINID address of the token holder
+     *  @param _balance the current balance of the token holder
+     *  Only the owner of the Compliance smart contract can call this function
+     *  emits a `IDBalancePreSet` event
+     */
+    function preSetModuleState(address _compliance, address _id, uint256 _balance) external {
+        require(OwnableUpgradeable(_compliance).owner() == msg.sender, "only compliance owner call");
+        require(!IModularCompliance(_compliance).isModuleBound(address(this)), "cannot do on bound compliance");
+        _preSetModuleState(_compliance, _id, _balance);
+        _compliancePresetStatus[_compliance] = true;
```

Comment:
i would set the preset status to true only when all ID balances have been set, let's add an additional function to set the preset status to true.